grapher/mapCharts/MapTooltip.tsx

```diff
@@ -2,113 +2,118 @@ import * as React from "react"
-            sparkBarValues,
-            (d) => d.time <= tooltipDatum.time
-        ).slice(-this.sparkBarsToDisplay)
+    @computed private get entityName(): EntityName {
+        return this.props.entityName
     }
 
-    @computed private get sparkBarsDomain(): [number, number] {
-        const lastVal = last(this.sparkBarsData)
-
-        const end = lastVal ? this.sparkBarsDatumXAccessor(lastVal) : 0
-        const start = end > 0 ? end - this.sparkBarsToDisplay + 1 : 0
+    // Table pre-filtered by targetTime, exlcudes time series
+    @computed private get mapTable(): OwidTable {
+        const table =
+            this.props.manager.transformedTable ?? this.props.manager.table
+        return table.filterByEntityNames([this.entityName])
+    }
 
-        return [start, end]
+    @computed private get timeSeriesTable(): OwidTable | undefined {
+        if (this.mapColumnSlug === undefined) return undefined
+        return this.props.timeSeriesTable
+            .filterByEntityNames([this.entityName])
+            .columnFilter(
+                this.mapColumnSlug,
+                isNumber,
+                "Drop rows with non-number values in Y column"
+            )
     }
 
-    @computed private get currentSparkBar(): number | undefined {
-        const lastVal = last(this.sparkBarsData)
-        return lastVal ? this.sparkBarsDatumXAccessor(lastVal) : undefined
+    @computed private get datum(): OwidVariableRow<PrimitiveType> | undefined {
+        return this.mapTable.get(this.mapColumnSlug).owidRows[0]
     }
 
-    colorScale = this.props.colorScale ?? new ColorScale()
+    @computed private get hasTimeSeriesData(): boolean {
+        return this.timeSeriesTable !== undefined
+            ? this.timeSeriesTable.numRows > 1
+            : false
+    }
 
-    @computed private get renderSparkBars(): boolean | undefined {
-        return this.props.manager.mapIsClickable
+    @computed private get lineColorScale(): ColorScale {
+        const oldManager = this.props.colorScaleManager
+        // Make sure all ColorScaleManager props are included.
+        // We can't ...rest here because I think mobx computeds aren't
+        // enumerable or something.
+        const newManager: NoUndefinedKeys<ColorScaleManager> = {
+            colorScaleConfig: oldManager.colorScaleConfig,
+            hasNoDataBin: oldManager.hasNoDataBin,
+            defaultNoDataColor: oldManager.defaultNoDataColor,
+            defaultBaseColorScheme: oldManager.defaultBaseColorScheme,
+            colorScaleColumn: oldManager.colorScaleColumn,
+            transformColor: darkenColorForLine,
+        }
+        return new ColorScale(newManager)
     }
 
-    @computed private get darkestColorInColorScheme(): string | undefined {
-        const { colorScale } = this
-        return colorScale.isColorSchemeInverted
-            ? first(colorScale.baseColors)
-            : last(colorScale.baseColors)
+    @computed private get textColorScale(): ColorScale {
```

Comment:
Hm, this is exactly the same as `lineColorScale`...
I like that you make the separation explicit, but maybe a simple `return this.lineColorScale` is enough unless and until we actually want them to differ?